sub-packages/bionemo-geometric/src/bionemo/geometric/atom_featurizers.py

```diff
@@ -0,0 +1,400 @@
+        return one_hot_enc(atom.GetTotalDegree(), self.n_dim)
+
+
+class ChiralTypeFeaturizer(BaseFeaturizer):
+    """Class for featurizing atom by its chirality type."""
+
+    def __init__(self) -> None:
+        """Initializes ChiralTypeFeaturizer class."""
+        pass
+
+    @property
+    def n_dim(self) -> int:
+        """Returns dimensionality of the computed features."""
+        return MAX_CHIRAL_TYPES
+
+    def get_features(self, atom: Atom) -> List[bool]:
+        """Returns features of the atom."""
+        return one_hot_enc(int(atom.GetChiralTag()), MAX_CHIRAL_TYPES)
+
+
+class TotalNumHFeaturizer(BaseFeaturizer):
+    """Class for featurizing atom by total number of hydrogens."""
+
+    def __init__(self) -> None:
+        """Initializes TotalNumHFeaturizer class."""
+        pass
+
+    @property
+    def n_dim(self) -> int:
+        """Returns dimensionality of the computed features."""
+        return MAX_NUM_HS
+
+    def get_features(self, atom: Atom) -> List[bool]:
+        """Returns features of the atom."""
+        return one_hot_enc(atom.GetTotalNumHs(), self.n_dim)
+
+
+class HybridizationFeaturizer(BaseFeaturizer):
+    """Class for featurizing atom by its hybridization type."""
+
+    def __init__(self) -> None:
+        """Initializes HybridizationFeaturizer class."""
+        pass
+
+    @property
+    def n_dim(self) -> int:
+        """Returns dimensionality of the computed features."""
+        return MAX_HYBRIDIZATION_TYPES
+
+    def get_features(self, atom: Atom) -> List[bool]:
+        """Returns features of the atom."""
+        return one_hot_enc(atom.GetHybridization(), self.n_dim)
+
+
+class AromaticityFeaturizer(BaseFeaturizer):
+    """Class for featurizing atom based on its aromaticity."""
+
+    def __init__(self) -> None:
+        """Initializes AromaticityFeaturizer class."""
+        pass
+
+    @property
+    def n_dim(self) -> int:
+        """Returns dimensionality of the computed features."""
+        return 1
+
+    def get_features(self, atom: Atom) -> List[bool]:
+        """Returns features of the atom."""
+        return [atom.GetIsAromatic()]
+
+
+class PeriodicTableFeaturizer(BaseFeaturizer):
+    """Class for featurizing atom by its position (period and group) in the periodic table."""
+
+    def __init__(self) -> None:
+        """Initializes PeriodicTableFeaturizer class."""
+        self.pt = Chem.GetPeriodicTable()
+
+    @property
+    def n_dim(self) -> int:
+        """Returns dimensionality of the computed features."""
+        return 25
+
+    def get_period(self, atom: Chem.Atom) -> List[bool]:
+        """Returns one-hot encoded period of atom."""
+        atomic_number = atom.GetAtomicNum()
+        # The number of elements per period in the periodic table
+        period_limits = [2, 10, 18, 36, 54, 86, 118]
+
+        # Determine the period based on atomic number
+        for period, limit in enumerate(period_limits, start=1):
```

Comment:
Would this return a -1 encoding for hydrogen or helium?

Reply:
No. `period` would be set to 1. Hydrogen and helium would be encoded as `[True, False, False, False, False, False, False]`

Reply:
Right, missed the start keyword, thanks